diff --git a/arduino/vacuum_pump/motor.cpp b/arduino/vacuum_pump/motor.cpp
index 89faded..b8732b4 100644
--- a/arduino/vacuum_pump/motor.cpp
+++ b/arduino/vacuum_pump/motor.cpp
@@ -28,9 +28,9 @@ void setup() {
   pinMode(TB6612_BIN2, OUTPUT);
 
   digitalWrite(TB6612_STBY, LOW);
-  analogWrite(TB6612_PWMA, MAXPWM);
+  digitalWrite(TB6612_PWMA, HIGH);
   digitalWrite(TB6612_AIN1, LOW);
-  digitalWrite(TB6612_AIN2, LOW);
+  analogWrite(TB6612_AIN2, MAXPWM);
   digitalWrite(TB6612_PWMB, HIGH);
   digitalWrite(TB6612_BIN1, LOW);
   digitalWrite(TB6612_BIN2, LOW);
@@ -55,18 +55,15 @@ void speed(int pwm, int port) {
   if (port == 0) { // port A
     pwma = pwm;
     pwma_percent = pwma * 100.0 / MAXPWM;
-    if (pwm == 0) {
+    // with motor pwm input pin connected to TB6612 pin AOUT1
+    if (pwm > 0) {
       digitalWrite(TB6612_AIN1, LOW);
-      digitalWrite(TB6612_AIN2, LOW);
-      analogWrite(TB6612_PWMA, pwm);
-    } else if (pwm > 0) {
-      digitalWrite(TB6612_AIN1, HIGH);
-      digitalWrite(TB6612_AIN2, LOW);
-      analogWrite(TB6612_PWMA, pwm);
+      digitalWrite(TB6612_PWMA, HIGH);
+      analogWrite(TB6612_AIN2, pwm);
     } else {
       digitalWrite(TB6612_AIN1, LOW);
-      digitalWrite(TB6612_AIN2, HIGH);
-      analogWrite(TB6612_PWMA, -pwm);
+      digitalWrite(TB6612_PWMA, HIGH);
+      analogWrite(TB6612_AIN2, 0);
     }
   } else { // port B
     pwmb = pwm;
